File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.py
from collections import defaultdict

def solution(s, skip, index):
    answer = ''
    dic = defaultdict(bool)
    for c in skip: dic[c] = True
    for c in s:
        cnt = 1
        temp_index = index
        while temp_index >= cnt:
            temp = (ord(c) - ord("a") + cnt) % 26
            if dic[chr(ord("a") + temp)]: temp_index += 1
            cnt += 1
        answer += chr(ord("a")+temp)
    return answer

File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.java
class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        String alpha = "abcdefghijklmnopqrstuvwxyz";
        int i, temp;
        
        for (char c: s.toCharArray()) {
            temp = index;
            i = alpha.indexOf(c);
            
            while (temp > 0) {
                if (skip.indexOf(alpha.charAt(++i % alpha.length())) == -1) {
                    temp--;
                }
            }
            answer += alpha.charAt(i % alpha.length());
        }
        return answer;
    }
}